.github/workflows/tests.yml

```diff
@@ -488,3 +485,54 @@ jobs:
                   needs.external_samplers.result != 'success' ||
                   needs.float32.result != 'success' ) }}
         run: exit 1
+
+
+  merge-reports:
+    runs-on: ubuntu-latest
+    needs: [all_tests, changes]
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Merge coverage reports
+        run: |
+          # Activate conda environment
+          conda activate pymc-test
+          # Find all coverage reports and merge them into a single file
+          python -m coverage combine --append coverage-*.xml
+
+      - name: Save combined coverage report
+        uses: actions/upload-artifact@v2
+        with:
+          name: coverage
+          path: coverage
```

Comment:
Why do we need this extra job? PyTensor doesn't seem to have an equivalent one.

Reply:
pytensor has just one job ('test_ubuntu') that generates the codecov report. in pymc there are many more, that generate partial reports. Adding step that combines them into one final report after all report-generating jobs finish seemed like an easier task

Reply:
by two steps do you mean merge-reports and upload coverage?